session/session_linux.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2018-2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package session
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/hashicorp/go-multierror"
+	"golang.org/x/sys/unix"
+)
+
+// Namespace assembles a NameSpace for this cpud, iff CPU_NONCE
+// is set and len(s.binds) > 0.
+// NOTE: this assumes we were started with Unshareflags set to CLONE_NEWNS.
```

Comment:
Can we assert this here? Or do we get an error somehow?
How will the user notice, what is "you will be sad" in terms of errors raised?

Reply:
it's very hard and probably impossible to test for. The tests currently will detect it, but there's NO WAY in linux to ensure you have an unshared name space that I have found. I've looked. This is more a warning to our future selves or users of this package. 

The main way you'd notice is if you and another user, sharing a mount, started stepping on each other. 

Note, however, that cpud does the right thing, by setting Unshareflags to CLONE_NEWNS, and I got a fix into the go runtime that makes sure that CLONE_NEWNS works correctly.

But it's not easy. Linux namespaces are a pretty terrible mess.

Reply:
Would you mind expanding the warning to our future selves to reflect this? I wouldn't have guessed. =)

Reply:
Is the new text any better?

Reply:
Sounds good, yea! Nit: `CLONE_NEWS` sould be `CLONE_NEWNS`.

Reply:
done, thanks.

Reply:
hope this works, would like to get this in and move on to the server package. Thanks for reviews :-)